app/assets/javascripts/angular/controllers/pages/recipe_finder_controller.js.erb

```diff
@@ -155,14 +128,21 @@ app.controller('RecipeFinderCtrl', ['$scope', '$http', '$routeParams', function(
     }
   };
 
+  $scope.searchWord = "Ingredient";
+
   $scope.addKeyword = function() {
     if ($scope.newKeyword != undefined && $scope.newKeyword != "" && $scope.newKeyword.length > 2) {
       if (checkDuplicates($scope.newKeyword,$scope.search_parameters.keywords) == true) {
```

Comment:
$scope.addKeyword = function() {
      if ($scope.newKeyword != undefined && $scope.newKeyword != "" && $scope.newKeyword.length > 2) {
        if (checkDuplicates($scope.newKeyword,$scope.search_parameters.keywords) == true || checkDuplicates($scope.newKeyword, $scope.search_parameters.ingredients) == true) {
          $scope.newKeyword = '';
          // need noty
        } else {
          if ($scope.searchWord == "Ingredient") {
            $scope.search_parameters.ingredients.push($scope.newKeyword);
            $scope.newKeyword = '';
          } else {
            $scope.search_parameters.keywords.push($scope.newKeyword);
            $scope.newKeyword = '';
          } 
        }
      } else {
        $scope.newKeyword = '';
        // needs notification of empty string
      }
      $scope.submitParams();
    }

you need to add the or to make this work